zio-http/src/main/scala/zhttp/http/auth/package.scala

```diff
@@ -0,0 +1,26 @@
+package zhttp.http
+
+package object auth extends AuthedRequestSyntax {
+  import Partials._
+
+  type HttpAuthenticationMiddleware[-R, +A] =
+    Http[R, HttpError, Request, AuthedRequest[A]]
+
+  object HttpAuthenticationMiddleware {
+
+    def ofM[A]: HttpAuthenticationMiddlewareMPartiallyApplied[A] =
+      new HttpAuthenticationMiddlewareMPartiallyApplied[A]
+
+    def of[A]: HttpAuthenticationMiddlewarePartiallyApplied[A] =
```

Comment:
```suggestion
    def custom[A]: HttpAuthenticationMiddlewarePartiallyApplied[A] =
```

Reply:
Later we can add vanilla support for various kinds of authentication mechanisms.